src/cryptonote_basic/cryptonote_format_utils.cpp

```diff
@@ -1012,6 +1012,21 @@ namespace cryptonote
     return true;
   }
   //---------------------------------------------------------------
+  bool tx_outs_checked_for_torsion(const transaction& tx)
+  {
+    for (const auto &o: tx.vout)
```

Comment:
Perhaps this `for` loop can be removed and just one iteration done? We already assert through other consensus rules that the output type has to be the same across all outputs in a transaction.

Reply:
Kept as is because this way this function doesn't rely on a distinct consensus rule. If that rule changes for some reason elsewhere, then this may break silently. I figure that safety is worth the iterations here.

Reply:
Kept as is because this way this function doesn't rely on a distinct consensus rule. If that rule changes for some reason elsewhere, then this may break silently. I figure that safety is worth the iterations here.